src/dawn/CodeGen/GridTools/GTCodeGen.cpp

```diff
@@ -465,6 +469,14 @@ GTCodeGen::generateStencilInstantiation(const StencilInstantiation* stencilInsta
           const auto& field = fields[accessorIdx];
           std::string paramName = stencilInstantiation->getNameFromAccessID(field.getAccessID());
 
+          // intialize map entry      
+          collectedExtents[paramName] = std::vector<int>(field.getExtents().getSize()*2);
```

Comment:
this I think is a bug, looks like you tried to compute the enclosing extent, but you are infact resetting the extent of each param for every stage

Reply:
Also no need to use std::vector, we normally try to use structs/classes that makes the internal data structures readable... here you can simply use Extents class. Also you have plenty of functionality in those classes. 
So for example you don need to compute the enclosing using max,min yourself. There are methods for that in the Extents